lang/cc65/Portfile

```diff
@@ -12,43 +14,40 @@ long_description \
                 cc65 is a complete cross development package for 65(C)02 \
                 systems, including a powerful macro assembler, a C compiler, \
                 linker, librarian and several other tools.
+license         CC-BY-SA 
 
-# Per http://www.cc65.org/ discontinued support in 2013, moved to github
-# No version tags on github yet
 homepage        http://cc65.github.io/cc65
-distfiles       cc65-sources-${version}.tar.bz2
-use_bzip2       yes
+distfiles       V${version}.tar.gz
 
-master_sites    http://cc65.oldos.net/ \
-                http://bj.spline.de/cc65/ \
-                ftp://ftp.musoftware.de/pub/uz/cc65/ \
-                ftp://ftp.musoftware.de/pub/uz/cc65/old/
+master_sites    git:https://github.com/cc65/cc65/archive
 
-checksums       rmd160  ad728cb61113af4592460bce7d42ca57f6a2fb1a \
-                sha256  a98a1b69d3fa15551fe7d53d5bebfc5f9b2aafb9642ee14b735587a421e00468
+checksums           rmd160  602e4956e4699103ab66dd654bfa53bce0766a85 \
+                    sha256  d14a22fb87c7bcbecd8a83d5362d5d317b19c6ce2433421f2512f28293a6eaab \
+                    size    2254374
 
 use_configure   no
 
-build.target    bins libs
-build.args      -f make/gcc.mak prefix=${prefix}
+build.target    bin lib
+build.args      PREFIX=${prefix}
 use_parallel_build no
 
-destroot.args   -f make/gcc.mak prefix=${destroot}${prefix}
-
-post-patch {
-    foreach file [glob ${worksrcpath}/src/*/make/gcc.mak] {
-        reinplace -E "s|^CC\[\[:space:\]\]*=\[\[:space:\]\]*gcc|CC=${configure.cc}|" $file
-    }
-}
+destroot.args   PREFIX=${prefix}
 
 variant docs description {Install extra documentation} {
     depends_build-append port:linuxdoc-tools
-    build.target-append docs
-    post-destroot {
-        set docdir ${destroot}${prefix}/share/doc/${name}
-        file mkdir "${docdir}/html"
-        foreach file [glob -directory ${docdir} *.htm*] {
-            move ${file} "${docdir}/html"
-        }
-    }
+    build.target-append doc
+
+	if {! [file exists ${prefix}/bin/perl] } {
+		ui_error "«${prefix}/bin/perl» is missing.
+
+The linuxdoc-tools depends on the perl executable.
+Please create an appropriate symbolic link.
+For example you could try using:
+
+pushd ${prefix}/bin
+sudo ln -s perl5.28 perl
+popd
+"
+		return -code error "missing dependency"
+	}
```

Comment:
Don't do this. /opt/local/bin/perl is provided by the perl5 port. If this port requires /opt/local/bin/perl and /usr/bin/perl is not sufficient, just add `depends_build-append port:perl5`. If either /usr/bin/perl or /opt/local/bin/perl is ok, use `depends_build-append bin:perl:perl5`. In fact, since your note says linuxdoc-tools is what requires this, make the change in the linuxdoc-tools port, not here. (It already has a build dependency, but if you're claiming it needs the dependency at runtime too, change it to depends_lib.)

Reply:
> Don't do this. /opt/local/bin/perl is provided by the perl5 port.

That didn't happen for me. Both perl 5.28 and perl 5.30 are installed and I have `/opt/local/bin/perl5` symlink but I had to create then `/opt/local/bin/perl` symlink manually.

If the perl symlink should have been created then this indicated a problem with the **perl5** port itself.

Not surprising considering that  **perl5** supports three perl version and MacPosts 8 different port versions and that there are several different ways to install perl (perl5, perl5.16, perl5.18, perl5.20, perl5.22, perl5.24, perl5.26, perl5.28, perl5.30) there could just be one installation route which won't create `/opt/local/bin/perl `. And it would be very difficult to find out which one.

I would suggest to reword the error message along the line of _“In rare cases «/opt/local/bin/perl » is not created …”_ making it clear that is should have happen. Removing the error will only lead to frustrations when the symlink is missing for whatever reason

>  just add depends_build-append port:perl5

Actually I did that. But since I already had two perl versions installed it made no difference so I deleted it again. But let me have a look at **linuxdoc-tools**.

```tlc
depends_build   bin:latex:texlive \
                path:bin/perl:perl5

depends_lib     port:flex \
                port:opensp \
                port:texinfo
```

Interesting. What is a “path:” dependency?  But yes, it should be a lib dependency.

Reply:
I deinstalled perl added the dependency and tried. And that worked. It's probably is a rare bug after all.

I didn't change linuxdoc-tools  as I think that changing multiple Portfiles in one pull request is something a second tier maintainer should do.

Reply:
I've fixed the linuxdoc-tools perl5 dependency in 8946a11ddc2e36b46de594975914f743cd7bb3a8 so let's remove the perl5 build dependency from cc65.

I have never heard of the /opt/local/bin/perl symlink disappearing before. The perl5 port has always unambiguously provided the /opt/local/bin/perl symlink. There is no codepath in the portfile that could lead to it not being installed. Maybe you inadvertently deleted it after perl5 was installed.

A dependency `path:a:b` means "if the file /opt/local/a does not exist, install port b". This syntax is used when there are multiple ports that could provide a file. For example, both graphviz and graphviz-devel provide /opt/local/bin/dot so the dependency is written `path:bin/dot:graphviz`. perl5 is the only port that provides /opt/local/bin/perl so there is no reason to use the `path:`-style when depending on perl5.